internal/am/delete_transfer.go

```diff
@@ -0,0 +1,38 @@
+package am
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+
+	"github.com/artefactual-sdps/enduro/internal/sftp"
+)
+
+const DeleteTransferActivityName = "DeleteTransferActivity"
+
+type DeleteTransferActivityParams struct {
+	Destination string
+}
+
+type DeleteTransferActivity struct {
+	client sftp.Client
+	logger logr.Logger
+}
+
+func NewDeleteTransferActivity(logger logr.Logger, client sftp.Client) *DeleteTransferActivity {
+	return &DeleteTransferActivity{client: client, logger: logger}
+}
+
+func (a *DeleteTransferActivity) Execute(ctx context.Context, params *DeleteTransferActivityParams) error {
+	a.logger.V(1).Info("Execute DeleteTransferActivity",
+		"Destination", params.Destination,
+	)
+
+	err := a.client.Delete(ctx, params.Destination)
+	if err != nil {
+		a.logger.V(1).Error(err, fmt.Sprintf("Delete transfer activity failed for %s", params.Destination))
```

Comment:
I think it's helpful to include the client error here (but use "%v" so it can't be unwrapped) to give more information about what went wrong.

Reply:
Client error is included in the error log. This is standard practice for the logger. Either way I will change this back to returning an error with fmt.Errorf()